### PR TITLE
Heartbeart and Deregistration

### PR DESCRIPTION
**Pull request by twoshark**
Bug Fixes:
- Peer Update/Membership Change

New Features:
- Heartbeat and auto deregistration on heartbeat failure

This PR follows https://github.com/dtp263/GoFire/pull/19 and will be kept a draft until that PR is approved/merged
This PR supersedes https://github.com/dtp263/GoFire/pull/18

### CODE REVIEW

File: src/routes/registration.go [hunk 1]
```diff
@@ -11,7 +12,7 @@ import (
 
 func (a *APIService) addRegistrationRoutes(e *echo.Echo) {
 	api := e.Group("/v1")
-	api.POST("/", a.peerUpdate)
+	api.POST("/peers", a.peerUpdate)
```
Comment (twoshark) [inline src/routes/registration.go:15]: I wasn't sure what to call this end point. /peers and /p2p are common in blockchains. /cluster and a bunch of other things would make sense too. I don't have a strong opinion on what we choose

File: src/cluster/master.go [hunk 2]
```diff
@@ -24,39 +26,50 @@ func (c *Cluster) KingMe() {
...
+					}
+				}
+			}
+		}
+	}()
 }
 
 //AddMicrocontroller attempts to add a microcontroller to the cluster and returns the response data. This should only be run by the master.
-func (c *Cluster) AddMicrocontroller(newMC mc.Config) (response PeerUpdateMessage, err error) {
+func (c *Cluster) AddMicrocontroller(newMC mc.Config) error {
```
Comment (twoshark) [inline src/cluster/master.go:57]: having the funcs that add and remove micros from the collection generate and send msgs felt wrong. Had to separate this work out

Comment (dtp263) [reply src/cluster/master.go:57]: Thats awesome. Totally agree

File: src/routes/registration.go (hunk 1 above)
Comment (dtp263) [reply src/routes/registration.go:15]: I like peers. Lets go with that

**Pull request merged**